packages/core/utils/src/dml/types.ts

```diff
@@ -108,3 +108,25 @@ export type Infer<T> = T extends DmlEntity<infer Schema>
         : Schema[K]["$dataType"]
     }>
   : never
+
+/**
+ * Extracts names of relationships from a schema
+ */
+export type ExtractEntityRelations<
+  Schema extends Record<string, any>,
+  OfType extends RelationshipTypes
+> = {
+  [K in keyof Schema & string]: Schema[K] extends RelationshipType<any>
+    ? Schema[K] extends { type: OfType }
+      ? K
+      : never
+    : never
+}[keyof Schema & string][]
```

Comment:
```suggestion
export type ExtractEntityRelations<
  Schema extends Record<string, any>,
  OfType extends RelationshipTypes
> = {
  [K in keyof Schema]: Schema[K] extends RelationshipType<any>
    ? Schema[K] extends { type: OfType }
      ? K
      : never
    : never
}
```

Reply:
Why did we remove the `[keyof Schema]` at the end after mapping the types 🤔